aten/src/ATen/functorch/DynamicLayer.cpp

```diff
@@ -166,10 +157,16 @@ static std::vector<DynamicLayer>& dynamicLayerStackAccessor() {
   return getRawFunctorchTLS()->dynamicLayerStack;
 }
 
-std::shared_ptr<bool> getLifeHandleForLevel(int64_t level) {
-  auto it = getGlobalDynmetaData().find(level);
-  TORCH_INTERNAL_ASSERT(it != kDynMetaDataSingleton.end(), "level should be alive");
-  return it->second;
+const std::shared_ptr<bool>& getLifeHandleForLevel(int64_t level) {
+  auto& dynamicLayerStack = dynamicLayerStackAccessor();
+  TORCH_INTERNAL_ASSERT(
+      dynamicLayerStack.size() >= level && level >= 1,
```

Comment:
just to sanity check this, why does this work now? Mostly not seeing the difference from the original PR and just want to check that there's not a local change that's lost here

Reply:
There is indeed a local change that was lost here, thanks for the catch